common/actions.js

```diff
@@ -92,9 +101,237 @@ export const deleteElements = (elements) => {
+    const contextTargetId = parseInt(potentialContextIds[2].split('+')[0].split(':')[0]) // userSaysId of intent2
+
+    // Insert context edges
+    edges.push({
+      group: "edges",
+      data: {
+        source: contextSourceId,
+        target: contextTargetId,
+        id: parseInt(potentialContextIds[0])},
+      classes: "r2us",
+    })
+  }
+
+  return edges
+}
+
+const buildCyIntent = (intent) => {
+// {  
+//    "id":"d8a056c9-829f-4852-a68e-26e52c08b2dc",
+//    "name":"40: u2+r1",
+//    "auto":false,
+//    "contexts":[  
+
+//    ],
+//    "templates":[  
+//       "u22",
+//       "u2"
+//    ],
+//    "userSays":[  
+//       {  
+//          "data":[  
+//             {  
+//                "text":"u2"
+//             }
+//          ],
+//          "isTemplate":true,
+//          "count":0,
+//          "created":-1
+//       },
+//       {  
+//          "data":[  
+//             {  
+//                "text":"u22"
+//             }
+//          ],
+//          "isTemplate":false,
+//          "count":0,
+//          "created":-1
+//       }
+//    ],
+//    "responses":[  
+//       {  
+//          "resetContexts":false,
+//          "action":"a1",
+//          "affectedContexts":[  
+//             {  
+//                "name":"40:u2+r1_36:u3+r2",
+//                "parameters":{  
+
+//                }
+//             },
+//             {  
+//                "name":"40:u2+r1_42:u4+r2",
+//                "parameters":{  
+
+//                }
+//             }
+//          ],
+//          "parameters":[  
+
+//          ],
+//          "speech":[  
+//             "r1",
+//             "r11"
+//          ]
+//       }
+//    ],
+//    "state":"LOADED",
+//    "priority":500000,
+//    "webhookUsed":false
+// }
+
+  // Intent name: 1:u1+3:e+2:r1 => userSaysId:text + edgeId:e + responseId:text
+  const potentialIds = intent.name.split('+')
+  const cyNodeIds = potentialIds.map(item => parseInt(item.split(':')[0]))
+  const response = intent.responses[0] 
+  const sppech = (response.speech instanceof Array) ? response.speech.join('\n'): response.speech
+
+  let cyIntentNodes = [
+    // Intent
+    {
+      group: "nodes",
+      data: { user_says: intent.templates.join('\n'), id: cyNodeIds[0] },
+      classes: "user_says",
+      position: {x: 100, y: 100},
+    },
+    {
+      group: "nodes",
+      data: { response: sppech, id: cyNodeIds[2], action: response.action },
+      classes: "response",
+      position: {x: 140, y: 100},
+    },
+    {
+      group: "edges",
+      data: {source: cyNodeIds[0], target: cyNodeIds[2], id: cyNodeIds[1]},
+      classes: "us2r",
+    }
+  ]
+
+  // Context
+  const incomingContexts = intent.contexts
+  const outgoingContexts = intent.responses[0].affectedContexts
+  cyIntentNodes = cyIntentNodes.concat(createContextEdges(incomingContexts, "incoming"), createContextEdges(outgoingContexts, "outgoing"))
+
+  return cyIntentNodes
```

Comment:
Yes, we can. I notices it but I wasn't planning to fix it since it'll still work. Feel free to fix this.